src/onegov/core/upgrade.py

```diff
@@ -482,17 +482,45 @@ def has_column(self, table: str, column: str) -> bool:
             table, schema=self.schema
         )}
 
-    def has_enum(self, enum: str) -> bool:
-        return self.session.execute(f"""
+    def has_constraint(
+        self,
+        table_name: str,
+        constraint_name: str,
+    ) -> bool:
+        schema_name = self.app.schema
+        query = text(
+            """
+            SELECT constraint_name
+            FROM information_schema.table_constraints
+            WHERE table_schema = :schema
+            AND table_name = :table_name
+            AND constraint_type = 'FOREIGN KEY'
+            AND constraint_name = :constraint_name
+            """
+        ).bindparams(
+            bindparam('schema', schema_name),
+            bindparam('table_name', table_name),
+            bindparam('constraint_name', constraint_name)
+        )
+        result = self.session.execute(query)
+        return result.scalar() is not None
+
+    def has_enum(self, enum_name: str) -> bool:
+        query = text("""
             SELECT EXISTS (
                 SELECT 1 FROM pg_type
-                WHERE typname = '{enum}'
-                  and typnamespace = (
+                WHERE typname = :enum_name
+                  AND typnamespace = (
                     SELECT oid FROM pg_namespace
-                    WHERE nspname = '{self.schema}'
+                    WHERE nspname = :schema
                   )
             )
-        """).scalar()
+        """).bindparams(
+            bindparam('schema', self.app.schema),
+            bindparam('enum_name', enum_name)
+        )
+        result = self.session.execute(query)
+        return result.scalar() is not None
```

Comment:
```suggestion
        return result.scalar()
```
`EXISTS` subqueries always return a boolean, so this check is wrong. You might want to consider using an `EXISTS` subquery for the other helper function as well, so you can just directly return `result.scalar()`